node_modules/oae-transfer/lib/internal/dao.js

```diff
@@ -0,0 +1,778 @@
+
+                                        // If no more editors, return the list of new managers
+                                        if(arrk.length-1 === k) {
+                                            AuthzAPI.updateRoles(element.id, editorsToUpdateRole, function(err, usersToInvalidate) {
+                                                if(err) {
+                                                    return callback(err);
+                                                }
+
+                                                // If no more elements / member / editors 
+                                                if(arrj.length-1 === j && arrk.length-1 === k && arri.length-1 === i) {
+                                                    // Return the list of new managers
+                                                    return callback(null, editorsToUpdateArray);
+                                                }
+                                            });
+                                        }
+                                    }); 
+                                }); 
+                            }); 
+                        }
+                        else {
+                            // If no more elements / member 
+                            if(arrj.length-1 === j && arri.length-1 === i) {
+                                // Return the list of new managers
+                                return callback(null, editorsToUpdateArray);
+                            }
+                        }
+                    }
+                });
+            });
+        }
+        else {
+            // If no more elements
+            if(arrj.length-1 === j) {
+                // Return the list of new managers
+                return callback(null, editorsToUpdateArray);
+            }
+        }
+    });
+};
+
+/**
+ * Update roles and Get array of roles modified
+ * 
+ * @param {Array}       authzRoles
+ * @param {String}      idUserTarget
+ * @param {String}      idUserOrigin
+ * @param {Function}    callback                
+ * @param {Array}       _userHashes
+ *
+ * @returns array of roles modified
+ * @api private used to make the transter
+ */
+var _getQueriesUpdateUserRoles = function(authzRoles, idUserTarget, idUserOrigin, callback, _userHashes) {
+
+    _userHashes = _userHashes || []
+    
+    // If no more authzroles, return list of modified roles
+    if (_.isEmpty(authzRoles)) {
+        return callback(null, _userHashes);
+    }
+
+    var authzRole = authzRoles.shift();
+    var memberToUpdate = {};
+    var memberToRemove = {};
+    var newAuthzRoles = {};
+
+    _doUpdate(authzRole, idUserTarget, idUserOrigin, function (res) {
+
+        if(res) {
+
+            memberToUpdate[idUserTarget] = authzRole.role;
+
+            // Update target user's role
+            AuthzAPI.updateRoles(authzRole.resourceId, memberToUpdate, function(err, usersToInvalidate) {
```

Comment:
This method can be used to update several principals at once, so we don't need to make separate calls to it for all of them?